resources/grafana/sources/rhacs-cluster-overview.json

```diff
@@ -1825,6 +1825,435 @@
         }
       ],
       "type": "table"
+    },
+    {
+      "collapsed": false,
+      "gridPos": {
+        "h": 1,
+        "w": 24,
+        "x": 0,
+        "y": 28
+      },
+      "id": 140,
+      "panels": [],
+      "title": "Cluster Violations",
+      "type": "row"
+    },
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "PBFA97CFB590B2093"
+      },
+      "description": "Shows occurrence intensity of SELinux AVC denials.",
```

Comment:
Why do you show both the SELinux denials twice? Just from looking at the plots, it seems the `SELinux Violations per minute` is sufficient (same for NetworkPolicy).

Reply:
The one is a violations-per-minute graph, the other one is a heatmap-style visualization. They are clearly related, but they highlight different aspects.

I am a little bit undecided which one I prefer, that's why currently there are both in the PR / the dashboard.

What I find problematic about the graph is that we are -- in my opinion -- not primarily interested in measuring violations per time as we would be for something like "requests per second". As I understand the situation we we primarily want to know if "violations have occurred" or not. In the heatmap every time unit with a positive number of violations is marked with a visible bar (of the same height). But, agreed, it might make a huge difference for criticality if there is just one violation or if there are 100 -- as an example. Hence, the heatmap also colours the bar, depending on the number of occurrences of violations.

The graph on the other hand could, depending on the distribution of violations, hide situations where only a small number of violations have occurred, if there are other situations where a huge number of violations has occurred. But we should be interested in analyzing both situations.

Reminder: My current assumption here is that we will reach a baseline of zero for these violations. We are currently seeing some violations from scanner, but these should be fixed at some point.

That being said, we can of course discuss further offline. Just wanted to provide a brief sketch of my reasoning here. Bottom line being for me: I see value in both visualizations, but if you are saying that we should only keep one of them, I guess I would vote in favor of the heatmap.

Reply:
Tbh I don't really understand what benefit the heatmap plot provides. In mind I would use a heatmap to visualize three-dimensional information - basically its purpose is to add an additional "heat" dimension to an otherwise already two-dimensional plot. But in this case we only have two dimensions (time, number of violations) to begin with, so I would prefer a simple time series.

Take for example this plot

<img width="747" alt="Screenshot 2024-04-23 at 22 28 50" src="https://github.com/stackrox/rhacs-observability-resources/assets/55607356/c9c43565-49ac-4954-8309-ccb31b2530fa">

At a glance, the only information I'm getting is when violations have occurred. There may be a difference in the color of the bars, but it's imperceptible to my eyes. Compare now to this view, which contains the same information about when violations occurred, but adds information about how many violations occurred at what time. Much more useful in my opinion. Note that the peak in the middle goes totally unnoticed in the first plot.
 
<img width="747" alt="Screenshot 2024-04-23 at 22 31 50" src="https://github.com/stackrox/rhacs-observability-resources/assets/55607356/6013d285-c262-4e37-93c1-76a8841d6e96">

> The graph on the other hand could, depending on the distribution of violations, hide situations where only a small number of violations have occurred, if there are other situations where a huge number of violations has occurred. But we should be interested in analyzing both situations.

In this case you will see a big bump in the time series immediately grabbing your attention (without having to know what color means "a lot of violations" in the heatmap). You will also see a small blimp that you could zoom in, it would only be obscured if the difference is many orders of magnitude. And if this is the main concern, the best plot would still be a time series but with a log scale.

Reply:
Discussed offline, I am fine with removing the heatmap visualizations.